### PR TITLE
Return correct params to variant ids

### PR DESCRIPTION
**Pull request by k1bs**
Add an additional line returning the viewable_ids to ensure assignment
to @image instance var.

### CODE REVIEW

**Comment by fastjames** [comment]
I have pushed a followup commit to this branch that adds a basic request spec. It's nice, but it does have one flaw. Right now, if I comment out your one-line fix the test still passes. Therefore, I don't think it fully replicates the problem you were having. Do you see any changes we could make to the test that might make it truer to life?

File: spec/controllers/spree/admin/images_controller_spec.rb [hunk 1]
```diff
@@ -18,6 +18,33 @@ module Admin
...
+        }
+        let(:create_params) {
+          {
+            image: {
+              viewable_ids: [variant.id],
+            },
+          }
+        }
+
+        before do
```
Comment (bradediger) [inline spec/controllers/spree/admin/images_controller_spec.rb]: This should probably be an `around` so we can restore its value, no?

File: app/controllers/spree/admin/images_controller_decorator.rb [hunk 2]
```diff
@@ -30,5 +30,8 @@ def viewable_ids
       unless params[:image][:viewable_ids].present?
         params[:image][:viewable_ids] = [@product.master.id]
       end
+
+      # return value to ensure assignment inside set_variants
+      params[:image][:viewable_ids]
```
Comment (bradediger) [inline app/controllers/spree/admin/images_controller_decorator.rb]: Is this method being used for its return value, its side effects on `params`, or both? It's doing a lot of stuff and it's not clear exactly what it's _supposed_ to be doing, which is a bit exacerbated by making it private and not unit-testing.

I know we're mostly constrained here to the controller/action shape dictated by upstream, but do we have any wiggle room to clean things up here? My eyes were glazing over a bit by the fourth or fifth time I read `params[:image][:viewable_ids]`. :-)

**Review by bradediger — COMMENTED** [review]
Left a bit of feedback inline regarding the decision to test this from the outside in. I think we may gain more flexibility if we can break apart this logic into smaller pieces so we can better test it, but I also understand that I'm working from a very limited view of the world here.

File: app/controllers/spree/admin/images_controller_decorator.rb (hunk 2 above)
Comment (fastjames) [reply app/controllers/spree/admin/images_controller_decorator.rb]: That's a good observation. I changed the method to only return the value and not have side-effects.

File: spec/controllers/spree/admin/images_controller_spec.rb (hunk 1 above)
Comment (fastjames) [reply spec/controllers/spree/admin/images_controller_spec.rb]: Yep, done.

**Comment by fastjames** [comment]
@bradediger I think I addressed both of your concerns.

**Review by bradediger — APPROVED** [review]
Yep, I like this implementation. Thanks!!

**Review by fastjames — APPROVED** [review]

**Pull request merged**